From aaeae96db2bb983376e75ff005860d0784ecf0a8 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Tue, 2 Jul 2024 11:21:03 +0200
Subject: [PATCH 1009/1695] fwk_io: fix GCC14 warning in fwk_io_vprintf()

warning: 'fwk_mm_alloc_notrap' sizes specified with 'sizeof' in the
earlier argument and not in the later argument [-Wcalloc-transposed-args]
  391 |     buffer = fwk_mm_alloc_notrap(sizeof(buffer[0]), (size_t)(length + 1));

SCP sync : I37ff47faca7b6fcbd629ed64f0e868f71f27a632

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c923f47ad44859fabc8a5aac2f3234340b4524d3 fwk_io: fix GCC14 warning in fwk_io_vprintf()]
Change-Id: I6774d159b0a5f31985781c7b6ef4941c0745f3db
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/390758
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/lib/scmi-server/SCP-firmware/framework/src/fwk_io.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/framework/src/fwk_io.c b/core/lib/scmi-server/SCP-firmware/framework/src/fwk_io.c
index e98387272..33350713e 100644
--- a/core/lib/scmi-server/SCP-firmware/framework/src/fwk_io.c
+++ b/core/lib/scmi-server/SCP-firmware/framework/src/fwk_io.c
@@ -388,7 +388,7 @@ int fwk_io_vprintf(
         return FWK_E_STATE;
     }
 
-    buffer = fwk_mm_alloc_notrap(sizeof(buffer[0]), (size_t)(length + 1));
+    buffer = fwk_mm_alloc_notrap((size_t)(length + 1), sizeof(buffer[0]));
     if (buffer == NULL) { /* Not enough memory for the string buffer */
         return FWK_E_NOMEM;
     }
-- 
2.47.2

